pkg/sensors/handler.go

```diff
@@ -129,29 +132,32 @@ func (h *handler) addTracingPolicy(op *tracingPolicyAdd) error {
 	if err != nil {
 		col.err = err
 		col.state = LoadErrorState
-		h.collections[op.ck] = &col
+		collections[op.ck] = &col
```

Comment:
you could simplify here and there now that you we use a pointer, you could insert the collection at its creation and then mutate it as we do in other methods (where you changed it in previous commits), so that we don't risk not to assign the pointer at a return point.

Reply:
ah nevermind, you do it https://github.com/cilium/tetragon/pull/2506/commits/6355329ee257ca5fbf487b926db4b48fb25de4ff.